database/src/main/java/com/firebase/ui/database/FirebaseListAdapter.java

```diff
@@ -50,11 +50,10 @@
  * @param <T> The class type to use as a model for the data contained in the children of the given Firebase location
  */
 public abstract class FirebaseListAdapter<T> extends BaseAdapter {
-
-    private final Class<T> mModelClass;
-    protected int mLayout;
     protected Activity mActivity;
-    FirebaseArray mSnapshots;
+    protected final Class<T> mModelClass;
```

Comment:
I'm more inclined to make `mLayout` private than to make `mModelClass` protected. The logic is the same: subclasses should not be modifying these fields as it may lead to unexpected behavior.